[AC-1117] Add manage permission

differsthecat opened the pull request:
## Type of change

<!-- (mark with an `X`) -->

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Add the 'Manage' column to the `CollectionUser` and `CollectionGroup` tables, and all models and Stored Procedures that use it as well.

## Code changes

Most of these changes are all the same. It is adding the Manage column to all models, entities, and stored procedures that use the `CollectionUser` and `CollectionGroup` fields.

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

gbubemismith reviewed (commented):
Nice work @differsthecat

differsthecat commented:
@gbubemismith I updated the migration script to use CREATE OR ALTER where I could. I had to leave the drops for the procedures that used the type that I updated, as I can not alter it.

gbubemismith reviewed (approved):
Thanks for the changes, @differsthecat. This looks good from my end.
